app/assets/javascripts/controllers/flavor_credentials/flavor_form_controller.js

```diff
@@ -0,0 +1,105 @@
+ManageIQ.angular.app.controller('flavorFormController', ['$scope', '$http', 'flavorId', 'miqService', 'API', function($scope, $http, flavorId, miqService, API) {
+  var vm = this;
+
+  var init = function() {
+    vm.afterGet = false;
+
+    vm.flavorModel = {
+      name: '',
+      ram: '',
+      vcpus: '',
+      disk: '',
+      swap: '',
+      rxtx_factor: '',
+      ems_id: '',
+    };
+
+    vm.model = 'flavorModel';
+    vm.ems_list = [];
+
+    $scope.newRecord = flavorId === 'new';
+
+    ManageIQ.angular.scope = vm;
+
+    $http.get('/flavor/ems_list')
+      .then(getEmsFormDataComplete)
+      .catch(miqService.handleFailure);
+
+    API.get('/api/providers?collection_class=ManageIQ::Providers::EmbeddedAutomationManager')
+      .then(getManagerResource)
+      .catch(miqService.handleFailure);
+  };
+
+  $scope.cancelClicked = function() {
+    miqService.sparkleOn();
+    var message = __('Add of Flavor cancelled by user.');
+    var url = '/flavor/show_list';
+    miqFlashLater({
+      message: message,
+      level: 'warning',
+    });
+    window.location.href = url;
+  };
+
+  $scope.addClicked = function() {
+    miqService.sparkleOn();
+    API.post('/api/flavors/', vm.flavorModel)
+      .then(getBack)
+      .catch(miqService.handleFailure);
+  };
+
+  function setForm() {
+    vm.modelCopy = angular.copy(vm.flavorModel);
+    vm.afterGet = true;
+  }
+
+  var getBack = function(response) {
+    var message = '';
+    var err = false;
+    if (response.hasOwnProperty('results')) {
```

Comment:
I guess the API really returns a 200 on error?

Can you perhaps separate this function into 2? (error and non-error) .. looks like this has too many `if (! err)` all around, and could be more readable as 2 functions.